compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4841,8 +4938,272 @@ public void visit(BLangTypedescExpr accessExpr, AnalyzerData data) {
         data.resultType = types.checkType(accessExpr, actualType, data.expType);
     }
 
+    public LinkedHashSet<BType> getBasicNumericTypesInFiniteType(BType referredType) {
```

Comment:
Isn't this logic similar to getBasicNumericTypesInUnionType? Please refactor the methods to avoid duplicating code.